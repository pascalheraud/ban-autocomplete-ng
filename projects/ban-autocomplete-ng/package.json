{
  "name": "ban-autocomplete-ng",
  "types": "public-api.d.ts",
  "version": "1.0.5",
  "repository": {
    "type": "git",
    "url": "https://github.com/pascalheraud/ban-autocomplete-ng"
  },
  "peerDependencies": {
    "@angular/common": "^19.1.0",
    "@angular/core": "^19.1.0"
  },
  "dependencies": {
    "tslib": "^2.3.0"
  },
  "sideEffects": false
}
