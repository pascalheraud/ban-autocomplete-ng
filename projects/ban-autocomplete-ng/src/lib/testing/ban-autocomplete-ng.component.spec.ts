import { ComponentFixture, TestBed } from '@angular/core/testing';

import { BanAutocompleteNgComponent } from '../ban-autocomplete-ng.component';
import {
  TranslateFakeLoader,
  TranslateLoader,
  TranslateModule,
  TranslateService,
} from '@ngx-translate/core';
import { By } from '@angular/platform-browser';
import { beforeEach, describe, expect, it, vi } from 'vitest';
import gre from './ban-autocomplete-ng.component.data';

const fetch = vi.fn();
global.fetch = fetch;

function createFetchResponse(data: any) {
  return { json: () => new Promise((resolve) => resolve(data)) };
}
describe('BanAutocompleteNgComponent', () => {
  let component: BanAutocompleteNgComponent;
  let fixture: ComponentFixture<BanAutocompleteNgComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [
        BanAutocompleteNgComponent,
        TranslateModule.forRoot({
          loader: {
            provide: TranslateLoader,
            useClass: TranslateFakeLoader,
          },
        }),
      ],
      providers: [TranslateService],
    }).compileComponents();

    fixture = TestBed.createComponent(BanAutocompleteNgComponent);
    component = fixture.componentInstance;
    fixture.autoDetectChanges();
  });

  it('should create', async () => {
    // Arrange : Mock fetch response
    fetch.mockResolvedValue(createFetchResponse(gre));

    // Assert : Snap empty state
    expect(component).toBeTruthy();
    expect(fixture.nativeElement.outerHTML).toMatchSnapshot('empty');

    // Act: Type "Gre" and fire update events
    const input = fixture.debugElement.query(By.css('input'));
    input.nativeElement.value = 'Gre';
    input.nativeElement.dispatchEvent(
      new Event('input', {
        bubbles: true,
        cancelable: true,
        composed: true,
      })
    );
    input.nativeElement.dispatchEvent(
      new KeyboardEvent('keydown', {
        key: 'G',
        bubbles: true,
        cancelable: true,
        composed: true,
      })
    );
    input.nativeElement.dispatchEvent(
      new KeyboardEvent('keyup', {
        key: 'G',
        bubbles: true,
        cancelable: true,
        composed: true,
      })
    );

    // Assert : wait for suggestions to be visible and snap them
    const suggestions = await vi.waitUntil(() =>
      fixture.debugElement.query(By.css('#suggestions.is-visible'))
    );
    expect(fixture.nativeElement.outerHTML).toMatchSnapshot('suggests');

    // Act : Click on the first suggestion
    const li = suggestions.query(By.css('li:nth-child(1) div'));
    li.nativeElement.click();

    // Assert : wait for input to be set to "Grenoble" and snap the result
    await vi.waitUntil(() => input.nativeElement.value == 'Grenoble');
    expect(fixture.nativeElement.outerHTML).toMatchSnapshot('clicked');
  });
});
