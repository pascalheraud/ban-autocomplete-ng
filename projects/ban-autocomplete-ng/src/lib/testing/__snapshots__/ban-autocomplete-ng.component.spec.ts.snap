// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BanAutocompleteNgComponent > should create > clicked 1`] = `
"<div id="root0" ng-version="19.2.1"><div class="ng-autocomplete"><ng-autocomplete class="ng-autocomplete" ng-reflect-data="[object Object],[object Object" ng-reflect-placeholder="Saisisez une adresse" ng-reflect-min-query-length="3" ng-reflect-search-keyword="label" ng-reflect-item-template="[object Object]" ng-reflect-not-found-template="[object Object]" ng-reflect-custom-filter="customFilter(items, query) {
 "><div aria-owns="suggestions suggestions-history" class="autocomplete-container" ng-reflect-ng-class="[object Object]" aria-expanded="false"><div class="input-container"><input type="text" aria-autocomplete="list" role="combobox" autocomplete="off" placeholder="Saisisez une adresse" ng-reflect-model="Grenoble" aria-label="Saisisez une adresse" class="ng-untouched ng-valid ng-dirty"><div class="x"><i aria-label="Close" class="material-icons">close</i></div><!--bindings={
  "ng-reflect-ng-if": "true"
}--><!--bindings={}--></div><div id="suggestions" role="listbox" class="suggestions-container is-visible" ng-reflect-ng-class="[object Object]"><!--bindings={
  "ng-reflect-ng-if": "false"
}--><ul><!--bindings={
  "ng-reflect-ng-for-of": ""
}--></ul></div><div id="suggestions-history" role="listbox" class="suggestions-container is-hidden" ng-reflect-ng-class="[object Object]"><!--bindings={
  "ng-reflect-ng-if": "false"
}--><ul><!--bindings={
  "ng-reflect-ng-for-of": ""
}--></ul></div><!--bindings={
  "ng-reflect-ng-if": "false"
}--></div><!--bindings={
  "ng-reflect-ng-if": "false"
}--></ng-autocomplete><!--container--><!--container--></div></div>"
`;

exports[`BanAutocompleteNgComponent > should create > empty 1`] = `
"<div id="root0" ng-version="19.2.1"><div class="ng-autocomplete"><ng-autocomplete class="ng-autocomplete" ng-reflect-data="" ng-reflect-placeholder="Saisisez une adresse" ng-reflect-min-query-length="3" ng-reflect-search-keyword="label" ng-reflect-item-template="[object Object]" ng-reflect-not-found-template="[object Object]" ng-reflect-custom-filter="customFilter(items, query) {
 "><div aria-owns="suggestions suggestions-history" class="autocomplete-container" ng-reflect-ng-class="[object Object]" aria-expanded="false"><div class="input-container"><input type="text" aria-autocomplete="list" role="combobox" autocomplete="off" placeholder="Saisisez une adresse" ng-reflect-model="" aria-label="Saisisez une adresse" class="ng-untouched ng-pristine ng-valid"><!--bindings={
  "ng-reflect-ng-if": ""
}--><!--bindings={}--></div><div id="suggestions" role="listbox" class="suggestions-container is-hidden" ng-reflect-ng-class="[object Object]"><!--bindings={
  "ng-reflect-ng-if": "false"
}--><ul><!--bindings={
  "ng-reflect-ng-for-of": ""
}--></ul></div><div id="suggestions-history" role="listbox" class="suggestions-container is-visible" ng-reflect-ng-class="[object Object]"><!--bindings={
  "ng-reflect-ng-if": "false"
}--><ul><!--bindings={
  "ng-reflect-ng-for-of": ""
}--></ul></div><!--bindings={
  "ng-reflect-ng-if": "false"
}--></div><!--bindings={
  "ng-reflect-ng-if": "false"
}--></ng-autocomplete><!--container--><!--container--></div></div>"
`;

exports[`BanAutocompleteNgComponent > should create > suggests 1`] = `
"<div id="root0" ng-version="19.2.1"><div class="ng-autocomplete"><ng-autocomplete class="ng-autocomplete" ng-reflect-data="[object Object],[object Object" ng-reflect-placeholder="Saisisez une adresse" ng-reflect-min-query-length="3" ng-reflect-search-keyword="label" ng-reflect-item-template="[object Object]" ng-reflect-not-found-template="[object Object]" ng-reflect-custom-filter="customFilter(items, query) {
 "><div aria-owns="suggestions suggestions-history" class="autocomplete-container active" ng-reflect-ng-class="[object Object]" aria-expanded="true"><div class="input-container"><input type="text" aria-autocomplete="list" role="combobox" autocomplete="off" placeholder="Saisisez une adresse" ng-reflect-model="Gre" aria-label="Saisisez une adresse" class="ng-untouched ng-valid ng-dirty"><div class="x"><i aria-label="Close" class="material-icons">close</i></div><!--bindings={
  "ng-reflect-ng-if": "true"
}--><!--bindings={}--></div><div id="suggestions" role="listbox" class="suggestions-container is-visible" ng-reflect-ng-class="[object Object]"><!--bindings={
  "ng-reflect-ng-if": ""
}--><ul><li class="item"><!--bindings={
  "ng-reflect-ng-if": "false"
}--><div><a>Grenoble</a><!--bindings={
  "ng-reflect-ng-template-outlet-context": "[object Object]"
}--></div><!--bindings={
  "ng-reflect-ng-if": "true"
}--></li><li class="item"><!--bindings={
  "ng-reflect-ng-if": "false"
}--><div><a>grenoble 85510 Le Boup√®re</a><!--bindings={
  "ng-reflect-ng-template-outlet-context": "[object Object]"
}--></div><!--bindings={
  "ng-reflect-ng-if": "true"
}--></li><li class="item"><!--bindings={
  "ng-reflect-ng-if": "false"
}--><div><a>Grenod 42800 Saint-Joseph</a><!--bindings={
  "ng-reflect-ng-template-outlet-context": "[object Object]"
}--></div><!--bindings={
  "ng-reflect-ng-if": "true"
}--></li><li class="item"><!--bindings={
  "ng-reflect-ng-if": "false"
}--><div><a>Grenoble 28400 Arcisses</a><!--bindings={
  "ng-reflect-ng-template-outlet-context": "[object Object]"
}--></div><!--bindings={
  "ng-reflect-ng-if": "true"
}--></li><li class="item"><!--bindings={
  "ng-reflect-ng-if": "false"
}--><div><a>Route de Grenoble 69800 Saint-Priest</a><!--bindings={
  "ng-reflect-ng-template-outlet-context": "[object Object]"
}--></div><!--bindings={
  "ng-reflect-ng-if": "true"
}--></li><li class="item"><!--bindings={
  "ng-reflect-ng-if": "false"
}--><div><a>Rue de Grenoble 31500 Toulouse</a><!--bindings={
  "ng-reflect-ng-template-outlet-context": "[object Object]"
}--></div><!--bindings={
  "ng-reflect-ng-if": "true"
}--></li><li class="item"><!--bindings={
  "ng-reflect-ng-if": "false"
}--><div><a>Avenue de Grenoble 93600 Aulnay-sous-Bois</a><!--bindings={
  "ng-reflect-ng-template-outlet-context": "[object Object]"
}--></div><!--bindings={
  "ng-reflect-ng-if": "true"
}--></li><!--bindings={
  "ng-reflect-ng-for-of": "[object Object],[object Object"
}--></ul></div><div id="suggestions-history" role="listbox" class="suggestions-container is-hidden" ng-reflect-ng-class="[object Object]"><!--bindings={
  "ng-reflect-ng-if": "false"
}--><ul><!--bindings={
  "ng-reflect-ng-for-of": ""
}--></ul></div><!--bindings={
  "ng-reflect-ng-if": "false"
}--></div><!--bindings={
  "ng-reflect-ng-if": "false"
}--></ng-autocomplete><!--container--><!--container--></div></div>"
`;
